src/shared/tpm2-util.c

```diff
@@ -480,6 +480,56 @@ void tpm2_pcr_mask_to_selection(uint32_t mask, uint16_t bank, TPML_PCR_SELECTION
         };
 }
 
+static void tpm2_log_debug_buffer(const void *buffer, const size_t size, const char *msg) {
+        if (!DEBUG_LOGGING || !buffer || !size)
+                return;
+
+        _cleanup_free_ char *h = hexmem(buffer, size);
+        if (h)
+                log_debug("%s: %s", msg ?: "Buffer", h);
+        else
+                log_oom();
+}
+
+static void tpm2_log_debug_digest(const TPM2B_DIGEST *digest, const char *msg) {
+        if (digest)
+                tpm2_log_debug_buffer(digest->buffer, digest->size, msg ?: "Digest");
+}
+
+static int tpm2_get_policy_digest(
+                Tpm2Context *c,
+                const Tpm2Handle *session,
+                TPM2B_DIGEST **ret_policy_digest) {
+
+        _cleanup_(Esys_Freep) TPM2B_DIGEST *policy_digest = NULL;
+        TSS2_RC rc;
+
+        assert(c);
+        assert(session);
+
+        if (DEBUG_LOGGING || ret_policy_digest) {
```

Comment:
Not a big matter, but I'd like to invert the condition and return earlier:
```c
if (!DEBUG_LOGGING && !ret_policy_digest)
        return 0;
```

Reply:
ack, updated